pkgs/tools/package-management/wapm/cli/default.nix

```diff
@@ -0,0 +1,52 @@
+{ darwin
+, fetchFromGitHub
+, lib
+, libiconv
+, openssl
+, rustPlatform
+, stdenv }:
+
+let
+  # Constants
+  pname = "wapm-cli";
+  owner = "wasmerio";
+  repo = "wapm-cli";
+
+  # Per-version variables
+  version = "0.5.5";
+  sha256 = "sha256-BKBd1tJwV4VOjRnAx/spQy3LIXzujrO2SS5eA1uybNA=";
+  cargoSha256 = "sha256-dv04AXOnzizjq/qx3qy524ylQHgE4gIBgeYI+2IRTug=";
+
+  # Helpers
+  inherit (lib) optionals;
+  inherit (rustPlatform) buildRustPackage;
+  inherit (stdenv) isDarwin;
+
+  # Build inputs for all systems
+  standardInputs = [ openssl ];
+
+  # Build inputs for macOS only
+  darwinInputs = optionals isDarwin [ libiconv ]
+    ++ (with darwin.apple_sdk.frameworks; [ Security SystemConfiguration ]);
+in buildRustPackage {
+  inherit pname version;
+
+  src = fetchFromGitHub {
+    inherit owner repo sha256;
+    rev = "v${version}";
+  };
```

Comment:
```suggestion
  src = fetchFromGitHub {
    owner = "wasmerio";
    repo = "wapm-cli";
    rev = "v${version}";
    sha256 = "sha256-BKBd1tJwV4VOjRnAx/spQy3LIXzujrO2SS5eA1uybNA=";
  };
```

Reply:
I'll make this change, but personally I'm not a fan of this seemingly standard pattern. I much prefer keeping values that change with each version in one place where it's clear to all what needs to change. With the suggested pattern here, there are no clear indicators of that. Furthermore, I avoid `rec` whenever possible and hope that Nixpkgs eventually comes to discourage its use.